arraykit.c

```diff
@@ -251,6 +261,195 @@ resolve_dtype_iter(PyObject *Py_UNUSED(m), PyObject *arg)
     return (PyObject *)AK_ResolveDTypeIter(arg);
 }
 
+
+//------------------------------------------------------------------------------
+// isin
+
+static PyObject *
+AK_isin_array_dtype_use_np(PyArrayObject *array, PyArrayObject *other, int assume_unique)
+{
+    PyObject* result = NULL;
+
+    PyObject* args = PyTuple_Pack(2, (PyObject*)array, (PyObject*)other);
+    AK_CHECK_NOT(args);
+
+    PyObject* kwarg = PyDict_New();
+    if (!kwarg) {
+        Py_DECREF(args);
+        return NULL;
+    }
+
+    PyObject* assume_unique_obj = PyLong_FromLong((long)assume_unique);
+    if (!assume_unique_obj) {
+        goto failure;
+    }
+
+    int success = PyDict_SetItemString(kwarg, "assume_unique", assume_unique_obj);
+    Py_DECREF(assume_unique_obj);
+    if (success == -1) {
+        goto failure;
+    }
+
+    PyObject* numpy = PyImport_ImportModule("numpy");
+    if (!numpy) {
+        goto failure;
+    }
+
+    PyObject* func = PyObject_GetAttrString(numpy, PyArray_NDIM(array) == 1 ? "in1d": "isin");
+    Py_DECREF(numpy);
+    if (!func) {
+        goto failure;
+    }
+
+    result = PyObject_Call(func, args, kwarg);
+    Py_DECREF(func);
+    if (!result) {
+        goto failure;
+    }
+
+    if (0) {
+    failure:
+        // These will always exist.
+        Py_DECREF(args);
+        Py_DECREF(kwarg);
+    }
```

Comment:
Sorry, my suggestion introduced a leak (in this case).
```suggestion
failure:
// These will always exist.
Py_DECREF(args);
Py_DECREF(kwarg);
```

You would use the `if (0) {`only to do the `Py_CLEAR(result)` step, had it been necessary. Example:

```c
if (0) {
failure:
    Py_CLEAR(result)
}
Py_DECREF(args);
Py_DECREF(kwarg);
return result;
```

Sorry about that.